packages/vite/src/node/plugins/worker.ts

```diff
@@ -305,7 +305,7 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
             workerConstructor === 'Worker'
               ? `${encodedJs}
           const decodeBase64 = (base64) => Uint8Array.from(atob(base64), c => c.charCodeAt(0));
-          const blob = typeof window !== "undefined" && window.Blob && new Blob([${
+          const blob = typeof self !== "undefined" && self.Blob && new Blob([${
```

Comment:
If this old code was executed in a Worker (i.e., when it is a nested worker), it will error because `window` does not exist and fall back to loading by data URI. Is that what was happening in your case? I want to clarify it because IIUC this won't happen in non-nested worker.

Reply:
Yes, the non-nested worker works fine. In a nested worker, the value of self.origin is 'null', which causes many APIs to be unusable due to the browser's same-origin policy.

Here is a simple code snippet to reproduce the error. Please try executing it in devtools.
```js
new Worker(
  "data:text/javascript;base64," + btoa(`
    console.log(111, self.origin === 'null')
    // test OPFS API
    navigator.storage.getDirectory() // => DOMException: Storage directory access is denied.
  `),
)
```